src/NuGet.Indexing/NuGetSearcherManager.cs

```diff
@@ -26,38 +27,65 @@ public NuGetSearcherManager(string indexName, Lucene.Net.Store.Directory directo
+        public IDictionary<string, Uri> RegistrationBaseAddress { get; }
+
+        /// <summary>Initializes a <see cref="NuGetSearcherManager"/> instance.</summary>
+        /// <param name="configuration">
+        /// The configuration to read which primarily determines whether the resulting instance will read from the local
+        /// disk or from blob storage.
+        /// </param>
+        /// <param name="directory">
+        /// Optionally, the Lucene directory to read the index from. If <c>null</c> is provided, the directory
+        /// implementation is determined based off of the configuration (<see cref="configuration"/>).
+        /// </param>
+        /// <param name="loader">
+        /// Optionally, the loader used to read the JSON data files. If <c>null</c> is provided, the loader
+        /// implementation is determined based off of the configuration (<see cref="configuration"/>).
+        /// </param>
+        /// <returns>The resulting <see cref="NuGetSearcherManager"/> instance.</returns>
+        public static NuGetSearcherManager Create(IConfiguration configuration, Lucene.Net.Store.Directory directory = null, ILoader loader = null)
         {
-            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(storagePrimary);
+            NuGetSearcherManager searcherManager;
+            var luceneDirectory = configuration.Get("Local.Lucene.Directory");
 
-            if (String.IsNullOrEmpty(indexContainer))
+            if (!string.IsNullOrEmpty(luceneDirectory))
             {
-                indexContainer = "ng-search-index";
-            }
+                string dataDirectory = configuration.Get("Local.Data.Directory");
 
-            if (String.IsNullOrEmpty(dataContainer))
+                searcherManager = new NuGetSearcherManager(
+                    luceneDirectory,
+                    directory ?? new SimpleFSDirectory(new DirectoryInfo(luceneDirectory)),
+                    loader ?? new FileLoader(dataDirectory));
+            }
+            else
             {
-                dataContainer = "ng-search-data";
+                string storagePrimary = configuration.Get("Storage.Primary");
+                string indexContainer = configuration.Get("Search.IndexContainer");
+                string dataContainer = configuration.Get("Search.DataContainer");
+
+                CloudStorageAccount storageAccount = CloudStorageAccount.Parse(storagePrimary);
+
+                if (string.IsNullOrEmpty(indexContainer))
+                {
+                    indexContainer = "ng-search-index";
+                }
+
+                if (string.IsNullOrEmpty(dataContainer))
+                {
+                    dataContainer = "ng-search-data";
+                }
+
+                searcherManager = new NuGetSearcherManager(
+                    indexContainer,
+                    directory ?? new AzureDirectory(storageAccount, indexContainer, new RAMDirectory()),
```

Comment:
Not scoped to this commit probably, but can we try the `SimpleFSDirectory` here as well instead of `RAMDirectory`? I think our app will be as fast as it is today but consume much less memory.